torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -120,10 +135,10 @@ PjRtComputationClient::PjRtComputationClient() {
     TF_VLOG(1) << "Initializing PjRt GPU client...";
     bool async = sys_util::GetEnvBool(env::kEnvPjrtAsyncGpuClient, true);
     int local_rank = sys_util::GetEnvInt(env::kEnvPjRtLocalRank, 0);
-    std::string dist_service_addr =
-        sys_util::GetEnvString(env::kEnvPjrtDistServiceAddr, "");
+    
+    int global_rank = getGlobalRank(local_rank);
     auto distributed_client =
-        MaybeInitializeDistributedRuntimeClient(local_rank, dist_service_addr);
+        MaybeInitializeDistributedRuntimeClient(global_rank);
     auto allowed_devices =
         std::make_optional<std::set<int>>(std::set{local_rank});
```

Comment:
We need to generalize this to support `CUDA_VISIBLE_DEVICES` and single-process-multi-device

Reply:
Yeah, we initially plan to incorporate CUDA_VISIBLE_DEVICES into this PR but we encountered some errors such as https://github.com/pytorch/xla/issues/5558#issuecomment-1756482280 and https://github.com/pytorch/xla/issues/5558#issuecomment-1751574292. We still plan to do it but probably in a follow-up pr